nimare/meta/mkda.py

```diff
@@ -832,6 +834,97 @@ def _compute_null_empirical(self, ma_maps, n_iters=10000):
             null_distribution[i_iter] = iter_ss_value
         self.null_distributions_["empirical_null"] = null_distribution
 
+    def _compute_null_analytic(self, ma_maps, n_foci_per_study):
+        """Compute uncorrected null distribution using analytic solution.
+
+        Parameters
+        ----------
+        ma_maps : list of imgs or numpy.ndarray
+            MA maps.
+
+        Notes
+        -----
+        This method adds two entries to the null_distributions_ dict attribute:
+        "histogram_bins" and "histogram_weights".
+        """
+        if isinstance(ma_maps, list):
+            ma_values = self.masker.transform(ma_maps)
+        elif isinstance(ma_maps, np.ndarray):
+            ma_values = ma_maps.copy()
+        else:
+            raise ValueError('Unsupported data type "{}"'.format(type(ma_maps)))
+
+        # Determine bins for null distribution histogram
+        # The maximum possible MA value for each study is the weighting factor (generally 1)
+        # times the number of foci in the study.
+        # To get the weighting factor, we find the minimum value in each MA map, ignoring zeros.
+        n_studies = ma_values.shape[0]
+        min_ma_values = np.zeros(n_studies)
+        for i_study in range(n_studies):
+            temp_ma_values = ma_values[i_study, :]
+            min_ma_values[i_study] = np.min(temp_ma_values[temp_ma_values != 0])
+
+        ma_scalar = np.mean(min_ma_values)
+        max_ma_values = min_ma_values * n_foci_per_study
+        max_poss_value = self._compute_summarystat(max_ma_values)
+        # Weighting is not supported yet, so I'm going to build my bins around the min MA value.
+        self.null_distributions_["histogram_bins"] = np.arange(
+            -ma_scalar / 2, max_poss_value + (ma_scalar * 1.5) + 0.001, ma_scalar
```

Comment:
Assuming a weight of 1 across the board, the histogram would look something like `[-0.5, 0.5, 1.5]`, with zeros going into the -0.5 to 0.5 bin, ones going into the 0.5 to 1.5 bin, etc.

Reply:
I don't think you need to use histograms here, since the values are always ints. You could use `np.unique()` with `return_counts=True`, which is probably also faster.

Reply:
The values may not be ints, depending on the `KDAKernel`'s `value` parameter and any weighting we might apply in the future (as in `MKDADensity`). Still, I don't think that impacts your point.

I'm not sure how to do what you're talking about, though. Would it be compatible with FWE if it's limited to _observed_ values, though? I know you mentioned dumping summary statistic values above the observed values into a final bin, but that would impact the way we implement the Monte Carlo procedure, right?

Reply:
Let me first say that I still think this is a rabbit hole that's not worth going down—I doubt anyone's going to use KDA, and if they do, they'll probably be doing themselves and/or the scientific community a disservice. There are almost no meta-analysis research questions I can think of where a fixed-effect estimate would be a good idea, so providing this as an easy-to-use tool seems likely to cause more harm than good.

With that said, what's the problem with having a larger number of bins than you might think you need? Even if the vast majority of values are ints, most of the probabilities will just get a probability of 0 and be ignored. it doesn't actually matter what bin the values end up in, AFAICS; remapping the observed values to p-values should still proceed just fine, no?

Reply:
> I doubt anyone's going to use KDA, and if they do, they'll probably be doing themselves and/or the scientific community a disservice.

I've added a warning to the docstring about this, and can add a `LGR.warning` as well.

> With that said, what's the problem with having a larger number of bins than you might think you need? Even if the vast majority of values are ints, most of the probabilities will just get a probability of 0 and be ignored. it doesn't actually matter what bin the values end up in, AFAICS; remapping the observed values to p-values should still proceed just fine, no?

I'm not sure, but shifting the bins seems to have solved the problem, so I'm happy to leave it as is.